redisson-tomcat/redisson-tomcat-8/src/main/java/org/redisson/tomcat/RedissonSession.java

```diff
@@ -65,7 +65,8 @@
     private final ReadMode readMode;
     private final UpdateMode updateMode;
 
-    private final AtomicInteger usages = new AtomicInteger();
+    // Initialize to one, since upon creation there is one usage
+    private final AtomicInteger usages = new AtomicInteger(1);
```

Comment:
Please clarify more about this change. In this case usages will be always 1 and not 0 when there are no usages.

Reply:
This is needed for handling requests which create a session. A request will come into the UsageValve, not hit `startUsage` because there is no session yet, continue onto the web application, create a session, and then when it exits the UsageValve it will try to call `endUsage` but because `usages` starts at 0, it will still not clear `loadedAttributes`

During creation we are exposing a reference to the session, meaning it is being used by someone, well after the UsageValve had a chance to increment the counter.

Alternatively, we could change the `endUsage` method to clear `loadedAttributes` when `usages` == -1, and if <0 reset to 0

Reply:
This change did cause issues in the case of multiple servers, and the 2nd server finds the session and still calls startUsage().
Added a commit with the change to endUsage() to handle the -1 case